nodeup/pkg/model/kubelet.go

```diff
@@ -156,6 +158,68 @@ func (b *KubeletBuilder) Build(c *fi.ModelBuilderContext) error {
 		return err
 	}
 
+	if kubeletConfig.CgroupDriver == "systemd" && b.Cluster.Spec.ContainerRuntime == "containerd" {
+		systemdPath := "/etc/systemd/system"
+
+		{
+			cgroup := kubeletConfig.KubeletCgroups
+			if cgroup != "" {
+				c.EnsureTask(b.buildCgroupService(cgroup))
+			}
+
+			manifest := &systemd.Manifest{}
+			manifest.Set("Service", "Slice", strings.Trim(cgroup, "/")+".slice")
+			manifestString := manifest.Render()
+			c.AddTask(&nodetasks.File{
+				Path:           path.Join(systemdPath, "containerd.service.d", "10-cgroup.conf"),
+				Type:           nodetasks.FileType_File,
+				Contents:       fi.NewStringResource(manifestString),
+				BeforeServices: []string{"containerd.service"},
+			})
+		}
+		{
+			cgroup := kubeletConfig.RuntimeCgroups
+			if cgroup != "" {
+				c.EnsureTask(b.buildCgroupService(cgroup))
+			}
+			manifest := &systemd.Manifest{}
+			manifest.Set("Service", "Slice", strings.Trim(cgroup, "/")+".slice")
+			manifestString := manifest.Render()
+			c.AddTask(&nodetasks.File{
+				Path:           path.Join(systemdPath, "kubelet.service.d", "10-cgroup.conf"),
```

Comment:
Why build drop-ins to unit files we already construct ourselves? Can't we just drop the slice configuration directly into containerd.service and kubelet.service?

Reply:
Good point. I'll see if I can fix that.